frontend/components/GesturesToSound.js

```diff
@@ -0,0 +1,168 @@
+    const [submitted, setSubmitted] = useState(false);
+    const [soundData, setSoundData] = useState(null);
+
+    const getCoords = (event) => {
+        if (!canvasRef.current) return;
+        const canvas = canvasRef.current;
+        const rect = canvas.getBoundingClientRect();
+        const scale = {
+            x: canvas.width / rect.width,
+            y: canvas.height / rect.height,
+        };
+        return {
+            x: (event.clientX - rect.left) * scale.x,
+            y: (event.clientY - rect.top) * scale.y,
+        };
+    };
+
+    const drawLine = (coords) => {
+        if (!canvasRef.current) return;
+        const canvas = canvasRef.current;
+        const context = canvas.getContext("2d");
+        if (context) {
+            context.beginPath();
+            context.lineWidth = 5;
+            context.lineJoin = "round";
+            context.lineCap = "round";
+            context.globalCompositeOperation = "source-over";
+            context.moveTo(coords[0].x, coords[0].y);
+            let i;
+            for (i = 0; i < coords.length - 1; i++) {
+                const c = (coords[i].x + coords[i + 1].x) / 2;
+                const d = (coords[i].y + coords[i + 1].y) / 2;
+                context.quadraticCurveTo(coords[i].x, coords[i].y, c, d);
+            }
+            context.stroke();
+        }
+    };
+
+    const beginDrawing = useCallback((event) => {
+        const coords = getCoords(event);
+        if (coords) {
+            setIsGesturing(true);
+            setCurrMouseCoords(prevCoords => [...prevCoords, coords]);
+        }
+    }, []);
+
+    const draw = useCallback((event) => {
+        if (isGesturing && !submitted.drawing) {
+            const newMouseCoord = getCoords(event);
+            if (newMouseCoord) {
+                setCurrMouseCoords(prevCoords => [...prevCoords, newMouseCoord]);
+                drawLine(currMouseCoords);
+            }
+        }
+    }, [isGesturing, currMouseCoords]);
+
+    const endDrawing = useCallback(() => {
+        setIsGesturing(false);
+        setCurrMouseCoords([]);
+    }, []);
+
+    useEffect(() => {
+        if (!canvasRef.current) return;
+        const canvas = canvasRef.current;
+        canvas.addEventListener("mousedown", beginDrawing);
+        canvas.addEventListener("mousemove", draw);
+        canvas.addEventListener("mouseup", endDrawing);
+        canvas.addEventListener("mouseleave", endDrawing);
+        if (!isGesturing && currMouseCoords.length) {
+            setAllMouseCoords(prevLines => [...prevLines, currMouseCoords]);
+        }
+        return () => {
+            canvas.removeEventListener("mousedown", beginDrawing);
+            canvas.removeEventListener("mousemove", draw);
+            canvas.removeEventListener("mouseup", endDrawing);
+            canvas.removeEventListener("mouseleave", endDrawing);
+        };
+    }, [beginDrawing, draw, endDrawing]);
+
+    const submitGesturesToAPI = () => {
```

Comment:
Check out the `fetchPost` function in `common.js`. It abstracts out a bunch of the fetch boilerplate, so please rewrite the innards of this function using that new function.=